### PR TITLE
fix: [PL-42963] add check in token name length

### PR DESCRIPTION
**Pull request by bharatgoelharness**
### Summary
Added the check in token name for max length 63 char
Screenshot 2023-11-22 at 9 57 01 AM
Screenshot 2023-11-22 at 9 56 45 AM

#### Scre
Screenshot 2023-11-22 at 9 59 23 AM
enshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/10-common/utils/Validation.tsx [hunk 1]
```diff
@@ -48,6 +48,30 @@ export function NameSchema(
...
+): Yup.Schema<string> {
+  return Yup.string()
+    .trim()
+    .required(config?.requiredErrorMsg ? config?.requiredErrorMsg : getString('common.validation.nameIsRequired'))
+    .matches(regexName, getString('common.validation.namePatternIsNotValid'))
+    .max(
+      MAX_VERSION_LABEL_LENGTH,
+      getString('common.validation.fieldCannotbeLongerThanN', {
+        name: getString('platform.delegates.tokens.tokensNameLength'),
+        n: MAX_VERSION_LABEL_LENGTH
```
Comment (AnupamIO) [inline src/modules/10-common/utils/Validation.tsx:63]: checked with Bharat, 63 is standard length throughout Harness, hence keeping the limit at 63. Approving, please merge after approval from UI eng.

**Review by AnupamIO — APPROVED** [review]

**Review by AnupamIO — DISMISSED** [review]

File: src/modules/10-common/utils/Validation.tsx [hunk 2]
```diff
@@ -48,6 +48,30 @@ export function NameSchema(
...
+      })
+    )
+}
+
+export function TokenNameSchema(
+  getString: UseStringsReturn['getString'],
+  config?: { requiredErrorMsg?: string }
+): Yup.Schema<string> {
+  return TokenNameSchemaWithoutHook(getString, config)
```
Comment (mayankVermaHarness) [inline src/modules/10-common/utils/Validation.tsx:72]: We don't need an extra method `<Action>WithoutHook`.
We can return the schema directly from here.

**Review by mayankVermaHarness — CHANGES_REQUESTED** [review]
We don't need an extra method `<Action>WithoutHook.`
We can return the schema directly from the method `TokenNameSchema`

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good Now
Please take review from Mayank also

**Review by mayankVermaHarness — APPROVED** [review]

**Pull request merged**